osu.Game.Rulesets.Mania/Beatmaps/LegacyBeatmapConverter.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Mania.Objects;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Mania.Beatmaps.Patterns;
+using osu.Game.Rulesets.Mania.MathUtils;
+
+namespace osu.Game.Rulesets.Mania.Beatmaps
+{
+    /// <summary>
+    /// Special converter used for converting from osu!stable beatmaps.
+    /// </summary>
+    internal class LegacyBeatmapConverter
+    {
+        private Pattern lastPattern = new Pattern();
+
+        private readonly FastRandom random;
+        private readonly Beatmap beatmap;
+
+        public LegacyBeatmapConverter(Beatmap beatmap)
+        {
+            this.beatmap = beatmap;
+
+            int seed = (int)Math.Round(beatmap.BeatmapInfo.Difficulty.DrainRate + beatmap.BeatmapInfo.Difficulty.CircleSize)
+                       * 20 + (int)(beatmap.BeatmapInfo.Difficulty.OverallDifficulty * 41.2) + (int)Math.Round(beatmap.BeatmapInfo.Difficulty.ApproachRate);
+            random = new FastRandom(seed);
+        }
+
+        public IEnumerable<ManiaHitObject> Convert(HitObject original)
+        {
+            var maniaOriginal = original as ManiaHitObject;
+            if (maniaOriginal != null)
+            {
+                yield return maniaOriginal;
+                yield break;
+            }
+
+            IEnumerable<ManiaHitObject> objects;
+            switch (beatmap.BeatmapInfo.RulesetID)
+            {
+                default:
+                    objects = generateConverted(original);
+                    break;
+                case 3:

```

Comment:
as discussed, we probably want to avoid having constants like this. instead making a method at ruleset `CreateBeatmapConstructor(bool isLegacy, bool isInCurrentRulesetFormat)`